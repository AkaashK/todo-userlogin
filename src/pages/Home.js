import React, { useState, useRef } from 'react';
import TodoList from './TodoList'
import History from './history'
import '../App.css'

function Home() {
    const [todolist, newTodo] = useState([])
    const todoName = useRef()

    function toggleTodo(id) {
        const newTodolist = [...todolist]
        const todo = newTodolist.find(todo => todo.id === id)
        todo.complete = !todo.complete
        newTodo(newTodolist)
    }

    function AddTodo() {
        const name = todoName.current.value
        if (name === '')
            return
        newTodo(prevtodos => {
            return [...prevtodos, { id: Date.now(), name: name, complete: false }]
        })
        todoName.current.value = null
    }

    function deleteTodos() {
        const newTodos = todolist.filter(todo => !todo.complete)
        newTodo(newTodos)
    }

    function logout() {
        History.push('/')
    }

    return (
        <div>
            <div>
                <h2>ToDo Dashboard</h2> 
                <button className="btn" type="button" onClick={logout}>logout</button>
            </div>
            <div className="App-header">
                <TodoList todolist={todolist} toggleTodo={toggleTodo} />
                <h3>Add todo...</h3>
                <input ref={todoName} type="text" /><br />
                <button className="btn" onClick={AddTodo}>Add Todo</button> <br />
                <button className="btn" onClick={deleteTodos}>Clear Completed Todos</button>
                <div>{todolist.filter(todo => !todo.complete).length} left to do</div>
            </div>
        </div>
    )
}

export default Home;

