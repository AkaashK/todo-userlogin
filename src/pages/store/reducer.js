const reducer = (state = [], action) => {
  switch (action.type) {
    case "Add-user":
      return [...state, action.data];
    default:
      return state;
  }
};

// export const activateUser = state => state.a

export default reducer;
