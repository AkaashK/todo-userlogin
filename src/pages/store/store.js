import { createStore } from 'redux';
import { devToolsEnhancer } from 'redux-devtools-extension'
import reducer from './reducer'
import * as actionCreators from './actions'

const store = createStore(reducer, devToolsEnhancer({ actionCreators, serialize: true, trace: true }));

export default store

