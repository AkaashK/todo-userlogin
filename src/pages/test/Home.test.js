import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import Login from "../Login";
import Home from '../Home'
import "@testing-library/jest-dom/extend-expect";
import { Provider } from 'react-redux'
import store from "../store/store"

afterEach(cleanup);
describe("Home", () => {
  it("renders correctly", () => {
    const { getByTestId } = render(<Login />);
    const emailid = getByTestId("emailTest");
    const passid = getByTestId("passTest");
    const btnid = getByTestId('button')
    
    fireEvent.change(emailid, {target: { value: "hello" }});
    fireEvent.change(passid, {target: { value: "123123" }});
    fireEvent.click(btnid)
    const tree = render(<Provider store={store}><Home email="hello"/></Provider>);
    expect(tree).toMatchSnapshot()
  })

  it("should call a function on click", () => {
    const mockAddTodo = jest.fn()
    const {getByTestId} = render(<Provider store={store}><Home email="hello"/></Provider>);
    const addoTodoBtn = getByTestId('Addtodo')
    fireEvent.click(addoTodoBtn)
    expect(addoTodoBtn.textContent).toBe('Add Todo')
    expect(mockAddTodo).toHaveBeenCalled()
  })

//   it('should store data from Add Todo', () => {
//     const {getByTestId} = render(<Provider store={store}><Home email="hello"/></Provider>);
//     const addTodoValue = getByTestId('addtodotest')
//     fireEvent.change(addTodoValue, {target: {value: 'Add a todo'}})
//     const tree = render(<Provider store={store}><Home email="hello"/></Provider>);
//     expect(tree.state()[0]).toMatch('Add a todo')
//   })
});