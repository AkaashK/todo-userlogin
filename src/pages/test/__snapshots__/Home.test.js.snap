// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App-header"
      >
        <img
          alt=""
          class="App-logo"
          src="logo.svg"
        />
        <h2
          class="title"
        >
          SIGN UP
        </h2>
        <p
          data-testid="alertmsg"
          id="alert"
        />
        email
        <input
          data-testid="emailTest"
          id="email"
          required=""
          type="text"
          value="hello"
        />
         
        <br />
        password
        <input
          data-testid="passTest"
          id="password"
          required=""
          type="password"
          value="123123"
        />
         
        <br />
        <button
          class="btn"
          data-testid="button"
          id="loginbtn"
          type="button"
        >
          signup
        </button>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h2>
            ToDo Dashboard
          </h2>
          <button
            class="btn"
            type="button"
          >
            logout
          </button>
        </div>
        <div
          class="App-header"
        >
          <h3>
            Add todo...
          </h3>
          <input
            data-testid="addtodotest"
            type="text"
          />
          <br />
          <button
            class="btn"
            data-testid="Addtodo"
          >
            Add Todo
          </button>
           
          <br />
          <button
            class="btn"
          >
            Clear Completed Todos
          </button>
          <div>
            0
             left to do
          </div>
          <button
            class="btn"
          >
            select all
          </button>
          <p>
            current user: 
            hello
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <h2>
          ToDo Dashboard
        </h2>
        <button
          class="btn"
          type="button"
        >
          logout
        </button>
      </div>
      <div
        class="App-header"
      >
        <h3>
          Add todo...
        </h3>
        <input
          data-testid="addtodotest"
          type="text"
        />
        <br />
        <button
          class="btn"
          data-testid="Addtodo"
        >
          Add Todo
        </button>
         
        <br />
        <button
          class="btn"
        >
          Clear Completed Todos
        </button>
        <div>
          0
           left to do
        </div>
        <button
          class="btn"
        >
          select all
        </button>
        <p>
          current user: 
          hello
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
